src/plugins/blocksmith/components/google-map-block-modal/index.jsx

```diff
@@ -74,24 +81,39 @@ export default class GoogleMapBlockModal extends React.Component {
   }
 
   setBlock() {
-    const { aside, hasUpdatedDate, mapType, q, updatedDate, zoom } = this.state;
+    const { aside, center, hasUpdatedDate, mapType, q, updatedDate, zoom } = this.state;
     const { block } = this.props;
+
     return block.schema().createMessage()
       .set('aside', aside)
+      .set('center', center ? GeoPoint.fromString(center) : null)
       .set('maptype', mapType || null)
       .set('q', q || null)
       .set('updated_date', hasUpdatedDate ? updatedDate : null)
       .set('zoom', zoom || null);
   }
 
-  handleAddBlock() {
+  async handleAddBlock() {
+    const { q } = this.state;
     const { onAddBlock, toggle } = this.props;
+
+    this.setState({ status: STATUS_PENDING });
+    const coordinates = await getGoogleMapCoordinates(q);
+    this.setState({ center: coordinates, status: STATUS_FULFILLED });
     onAddBlock(this.setBlock());
     toggle();
   }
 
-  handleEditBlock() {
+  async handleEditBlock() {
+    const { oldQ, q, center } = this.state;
     const { onEditBlock, toggle } = this.props;
+
+    if (q !== oldQ || !center) {
```

Comment:
@bhsiao55 - if the _oldQ_ is only used to compare with the initial value, then let's remove it and get that value directly from the block like this:

```
const { q, center } = this.state;
const { block, onEditBlock, toggle } = this.props;

if (q !== block.get('q') || !center) {
```

Reply:
yes that works out better